docker/upgrade-connector.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env sh
+set -eu -o pipefail
+
+connector_path="${HASURA_PLUGIN_CONNECTOR_CONTEXT_PATH:-/functions}"
+target_connector_version="$(cat /scripts/CONNECTOR_VERSION)"
+
+rm -rf /tmp/connector-upgrade
+mkdir /tmp/connector-upgrade
+
+# We copy the package.json and package-lock.json to a temporary directory
```

Comment:
Had a thought about this, did you try running `npm install --package-lock-only`? https://docs.npmjs.com/cli/v11/commands/npm-install#package-lock-only

Reply:
(I suppose you'd still need to do the `cat` stuff below to change the `package.*` files without messing up their perms)

Reply:
That's a good suggestion, I didn't know about that flag. I'll give it a shot, it's worth it even if it just improves performance.

Reply:
Looks like I can avoid all the copying and cat stuff with `--package-lock-only`. npm doesn't change the permissions on the files and doesn't modify node_modules. Good call dude!